### PR TITLE
Add support for Rocky8, Rocky9, Ubuntu build

### PR DESCRIPTION
**Pull request by velomatt**
- Add dependency installation scripts to support CentOS 7, Rocky8, Rocky9, and Ubuntu 22 build
- Update README to reflect changes
- Minor changes to support newer GCC compiler
- Add dockerfiles for each supported distro / OS

### CODE REVIEW

**Review by snubatreus — DISMISSED** [review]
Looks good.

**Review by snubatreus — APPROVED** [review]
Looks good

**Comment by ramachaitanyak** [comment]
NKV cli changes look good to me.

File: host/src/include_private/nkv_framework.h [hunk 1]
```diff
@@ -281,11 +282,17 @@
...
       if (nkv_in_memory_exec) {
-        data_cache = new std::unordered_map<std::string, nkv_value_wrapper*> [nkv_listing_cache_num_shards];
+        data_cache = new std::unordered_map<std::string, nkv_value_wrapper*> (nkv_listing_cache_num_shards);
+      }
+      cnt_cache.resize(nkv_read_cache_shard_size);
+      for (auto i=0; i<nkv_read_cache_shard_size; i++) {
+          // Initialize class object
+          nkv_lruCache<std::string, nkv_value_wrapper> *cache_obj
+              = new nkv_lruCache<std::string, nkv_value_wrapper>(nkv_read_cache_size);
+          cnt_cache.push_back(cache_obj);
```
Comment (ramachaitanyak) [inline host/src/include_private/nkv_framework.h:294]: Instead of pushing back we just need to access cnt_cache[i].

**Pull request merged**